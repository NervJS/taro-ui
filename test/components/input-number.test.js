import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'
import AtInputNumber from '../../.temp/components/input-number/index'

describe('AtInputNumber Snap', () => {
  it('render initial AtInputNumber', () => {
    const component = renderToString(<AtInputNumber />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInputNumber -- props customStyle', () => {
    const component = renderToString(<AtInputNumber customStyle='color:red;' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInputNumber -- props className', () => {
    const component = renderToString(<AtInputNumber className='test' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInputNumber -- props type=number', () => {
    const component = renderToString(<AtInputNumber type='number' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInputNumber -- props type=digit', () => {
    const component = renderToString(<AtInputNumber type='digit' />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInputNumber -- props disabled', () => {
    const component = renderToString(<AtInputNumber disabled />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInputNumber -- props disabledInput', () => {
    const component = renderToString(<AtInputNumber disabledInput />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInputNumber -- props value', () => {
    const component = renderToString(<AtInputNumber value={2} />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInputNumber -- props width', () => {
    const component = renderToString(<AtInputNumber width={200} />)
    expect(component).toMatchSnapshot()
  })

  it('render AtInputNumber -- props size', () => {
    const component = renderToString(<AtInputNumber size='large' />)
    expect(component).toMatchSnapshot()
  })
})
