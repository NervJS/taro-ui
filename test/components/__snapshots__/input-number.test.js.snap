// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtInputNumber Snap render AtInputNumber -- props className 1`] = `"<div class=\\"at-input-number test\\"><div class=\\"at-input-number__btn\\"><span class=\\"taro-text at-icon at-icon-subtract at-input-number__btn-subtract\\"></span></div><input style=\\"width:;\\" value=\\"1\\" class=\\"weui-input at-input-number__input\\" type=\\"number\\"/><div class=\\"at-input-number__btn\\"><span class=\\"taro-text at-icon at-icon-add at-input-number__btn-add\\"></span></div></div>"`;

exports[`AtInputNumber Snap render AtInputNumber -- props customStyle 1`] = `"<div style=\\"color:red;\\" class=\\"at-input-number\\"><div class=\\"at-input-number__btn\\"><span class=\\"taro-text at-icon at-icon-subtract at-input-number__btn-subtract\\"></span></div><input style=\\"width:;\\" value=\\"1\\" class=\\"weui-input at-input-number__input\\" type=\\"number\\"/><div class=\\"at-input-number__btn\\"><span class=\\"taro-text at-icon at-icon-add at-input-number__btn-add\\"></span></div></div>"`;

exports[`AtInputNumber Snap render AtInputNumber -- props disabled 1`] = `"<div class=\\"at-input-number\\"><div class=\\"at-input-number__btn at-input-number--disabled\\"><span class=\\"taro-text at-icon at-icon-subtract at-input-number__btn-subtract\\"></span></div><input style=\\"width:;\\" value=\\"1\\" class=\\"weui-input at-input-number__input\\" disabled type=\\"number\\"/><div class=\\"at-input-number__btn at-input-number--disabled\\"><span class=\\"taro-text at-icon at-icon-add at-input-number__btn-add\\"></span></div></div>"`;

exports[`AtInputNumber Snap render AtInputNumber -- props disabledInput 1`] = `"<div class=\\"at-input-number\\"><div class=\\"at-input-number__btn\\"><span class=\\"taro-text at-icon at-icon-subtract at-input-number__btn-subtract\\"></span></div><input style=\\"width:;\\" value=\\"1\\" class=\\"weui-input at-input-number__input\\" disabled type=\\"number\\"/><div class=\\"at-input-number__btn\\"><span class=\\"taro-text at-icon at-icon-add at-input-number__btn-add\\"></span></div></div>"`;

exports[`AtInputNumber Snap render AtInputNumber -- props size 1`] = `"<div class=\\"at-input-number at-input-number--lg\\"><div class=\\"at-input-number__btn\\"><span class=\\"taro-text at-icon at-icon-subtract at-input-number__btn-subtract\\"></span></div><input style=\\"width:;\\" value=\\"1\\" class=\\"weui-input at-input-number__input\\" type=\\"number\\"/><div class=\\"at-input-number__btn\\"><span class=\\"taro-text at-icon at-icon-add at-input-number__btn-add\\"></span></div></div>"`;

exports[`AtInputNumber Snap render AtInputNumber -- props type=digit 1`] = `"<div class=\\"at-input-number\\"><div class=\\"at-input-number__btn\\"><span class=\\"taro-text at-icon at-icon-subtract at-input-number__btn-subtract\\"></span></div><input style=\\"width:;\\" value=\\"1\\" class=\\"weui-input at-input-number__input\\" type=\\"number\\"/><div class=\\"at-input-number__btn\\"><span class=\\"taro-text at-icon at-icon-add at-input-number__btn-add\\"></span></div></div>"`;

exports[`AtInputNumber Snap render AtInputNumber -- props type=number 1`] = `"<div class=\\"at-input-number\\"><div class=\\"at-input-number__btn\\"><span class=\\"taro-text at-icon at-icon-subtract at-input-number__btn-subtract\\"></span></div><input style=\\"width:;\\" value=\\"1\\" class=\\"weui-input at-input-number__input\\" type=\\"number\\"/><div class=\\"at-input-number__btn\\"><span class=\\"taro-text at-icon at-icon-add at-input-number__btn-add\\"></span></div></div>"`;

exports[`AtInputNumber Snap render AtInputNumber -- props value 1`] = `"<div class=\\"at-input-number\\"><div class=\\"at-input-number__btn\\"><span class=\\"taro-text at-icon at-icon-subtract at-input-number__btn-subtract\\"></span></div><input style=\\"width:;\\" value=\\"2\\" class=\\"weui-input at-input-number__input\\" type=\\"number\\"/><div class=\\"at-input-number__btn\\"><span class=\\"taro-text at-icon at-icon-add at-input-number__btn-add\\"></span></div></div>"`;

exports[`AtInputNumber Snap render AtInputNumber -- props width 1`] = `"<div class=\\"at-input-number\\"><div class=\\"at-input-number__btn\\"><span class=\\"taro-text at-icon at-icon-subtract at-input-number__btn-subtract\\"></span></div><input style=\\"width:4.2667rem;\\" value=\\"1\\" class=\\"weui-input at-input-number__input\\" type=\\"number\\"/><div class=\\"at-input-number__btn\\"><span class=\\"taro-text at-icon at-icon-add at-input-number__btn-add\\"></span></div></div>"`;

exports[`AtInputNumber Snap render initial AtInputNumber 1`] = `"<div class=\\"at-input-number\\"><div class=\\"at-input-number__btn\\"><span class=\\"taro-text at-icon at-icon-subtract at-input-number__btn-subtract\\"></span></div><input style=\\"width:;\\" value=\\"1\\" class=\\"weui-input at-input-number__input\\" type=\\"number\\"/><div class=\\"at-input-number__btn\\"><span class=\\"taro-text at-icon at-icon-add at-input-number__btn-add\\"></span></div></div>"`;
